cirq-core/cirq/sim/act_on_state_vector_args.py

```diff
@@ -171,22 +171,15 @@ def _act_on_fallback_(
         qubits: Sequence['cirq.Qid'],
         allow_decompose: bool = True,
     ) -> bool:
-        strats = [
-            _strat_act_on_state_vector_from_apply_unitary,
-            _strat_act_on_state_vector_from_mixture,
-            _strat_act_on_state_vector_from_channel,
-        ]
+        if self._strat_act_on_state_vector_from_apply_unitary(action, qubits):
+            return True
+        if self._strat_act_on_state_vector_from_mixture(action, qubits):
+            return True
+        if self._strat_act_on_state_vector_from_channel(action, qubits):
+            return True
         if allow_decompose:
-            strats.append(strat_act_on_from_apply_decompose)
-
-        # Try each strategy, stopping if one works.
-        for strat in strats:
-            result = strat(action, self, qubits)
-            if result is False:
-                break  # coverage: ignore
-            if result is True:
+            if strat_act_on_from_apply_decompose(action, self, qubits):
                 return True
-            assert result is NotImplemented, str(result)
```

Comment:
It looks like the False / NotImplemented distinction could be removed because we never use False - i.e. a strategy can never unequivocally say "I failed, and all other strategies will also fail."

Was that the intent? It doesn't seem too difficult to add back if we find a need for it later, just wanted to confirm since it wasn't mentioned in the description.

Reply:
Yes, this was intentional. There was nowhere that was failing by force. I think ultimately that should be a top-level thing if anything anyway. The first section in act_on_fallback for any particular ActOnArgs should test whether the action can be applied before even trying a strategy. But yes it can be added back for any particular strategy if needed.